packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -642,56 +713,199 @@ abstract class TableBase extends Resource implements ITable {
 
   /**
    * Metric for the consumed read capacity units this table
-   *
-   * @default sum over a minute
    */
   public metricConsumedReadCapacityUnits(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
     return this.metric('ConsumedReadCapacityUnits', { statistic: 'sum', ...props });
   }
 
   /**
    * Metric for the consumed write capacity units this table
-   *
-   * @default sum over a minute
    */
   public metricConsumedWriteCapacityUnits(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
     return this.metric('ConsumedWriteCapacityUnits', { statistic: 'sum', ...props });
   }
 
   /**
    * Metric for the system errors this table
-   *
-   * @default sum over a minute
    */
-  public metricSystemErrors(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
-    return this.metric('SystemErrors', { statistic: 'sum', ...props });
+  public metricSystemErrors(props?: OperationalMetricOptions): cloudwatch.IMetric {
+
+    if (props?.dimensions?.Operation && props.operations) {
+      // these two represent the same thing, opted to require just one of them instead of dealing with merges
+      throw new Error('Operation can only be specified using the "operations" property, not the Operation dimension.');
+    }
+
+    const operations = [];
+
+    if (props?.dimensions?.Operation) {
+      // user specified the operation dimension, use it.
+      // note this means that props.operations is empty because of the above validation.
+      operations.push(props.dimensions?.Operation);
+    }
+
+    if (props?.operations) {
+      // user specified the operations property, use it.
+      // note this means that the Operation dimension is not passed because of the above validation.
+      operations.push(...props.operations);
+    }
+
+    if (operations.length == 0) {
+      // nor the Operation dimension nor the operations property was passed.
+      // default to using all operations.
+      operations.push(...Object.values(Operation));
+    }
+
+    const usingMetrics: Record<string, cloudwatch.IMetric> = {};
+
+    for (const operation of operations) {
```

Comment:
Can we move all the code around creating a MathExpression on multiple operations into a helper function? I find it very distracting here, and doesn't it occur in multiple places?

Reply:
I love that that you remind me of stuff I wanted to do and forgot :)